### PR TITLE
Issue1518 cooling tower

### LINKED ISSUE #1518: Cooling tower control sequence
This issue is to clean up the cooling tower control sequence.

### PR DESCRIPTION
**Pull request by JayHuLBL**
This closes #1518.

### CODE REVIEW

File: Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/Staging/Validation/Controller.mo [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    final width=0.18, final period=3600) "Boolean pulse"
+    annotation (Placement(transformation(extent={{-100,-60},{-80,-40}})));
+  Buildings.Controls.OBC.CDL.Logical.Not towStaUp "Tower stage up status"
+    annotation (Placement(transformation(extent={{-60,-60},{-40,-40}})));
+  Buildings.Controls.OBC.CDL.Logical.Sources.Pulse wseSta(
+    final width=0.15,
+    final period=3600,
+    final startTime=300) "Boolean pulse"
+    annotation (Placement(transformation(extent={{-60,50},{-40,70}})));
+  Buildings.Controls.OBC.CDL.Conversions.BooleanToInteger chiSta "Chiller stage "
```
Comment (milicag) [inline Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/Staging/Validation/Controller.mo]: space extra - please delete

File: Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/Controller.mo [hunk 2]
```diff
@@ -0,0 +1,494 @@
...
+  parameter Modelica.SIunits.Time TdWSE=0.1 "Time constant of derivative block"
+    annotation (Dialog(tab="Fan speed", group="WSE enabled",
+                       enable=hasWSE and (chiWatCon==Buildings.Controls.OBC.CDL.Types.SimpleController.PD or
+                                          chiWatCon==Buildings.Controls.OBC.CDL.Types.SimpleController.PID)));
+
+  // Fan speed control: controlling condenser return water temperature when WSE is not enabled
+  parameter Modelica.SIunits.TemperatureDifference LIFT_min[nChi]={12,12} "Minimum LIFT of each chiller"
+    annotation (Dialog(tab="Fan speed", group="Return temperature control"));
+  parameter Modelica.SIunits.Time iniPlaTim=600
+    "Time to hold return temperature to initial setpoint after plant being enabled"
```
Comment (milicag) [inline Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/Controller.mo]: `at` instead of `to` maybe?

File: Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/FanSpeed/Controller.mo [hunk 3]
```diff
@@ -0,0 +1,371 @@
...
+    "Time to hold return temperature to initial setpoint after plant being enabled"
+    annotation (Dialog(tab="Return temperature control", group="Setpoint"));
+  parameter Modelica.SIunits.Time ramTim=180
+    "Time to ramp return water temperature from initial value to setpoint"
+    annotation (Dialog(tab="Return temperature control", group="Setpoint"));
+  parameter Modelica.SIunits.Temperature TConWatRet_nominal[nChi]={303.15,303.15}
+    "Condenser water return temperature (condenser leaving) of each chiller"
+    annotation (Dialog(tab="Return temperature control", group="Setpoint"));
+  parameter Modelica.SIunits.Temperature TChiWatSupMin[nChi]={278.15,278.15}
+    "Lowest chilled water supply temperature oc each chiller"
```
Comment (milicag) [inline Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/FanSpeed/Controller.mo]: `of`

File: Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/FanSpeed/Controller.mo [hunk 4]
```diff
@@ -0,0 +1,371 @@
...
+    final unit=fill("1", nChi))
+    "Maximum cooling tower speed setpoint from each chiller head pressure control loop"
+    annotation (Placement(transformation(extent={{-140,-40},{-100,0}}),
+      iconTransformation(extent={{-140,-40},{-100,0}})));
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanInput uTowSta[nTowCel]
+    "Cooling tower cell operating status: true=running tower cell"
+    annotation (Placement(transformation(extent={{-140,-80},{-100,-40}}),
+      iconTransformation(extent={{-140,-70},{-100,-30}})));
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanInput uPla
+    "Plant enabling status"
```
Comment (milicag) [inline Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/FanSpeed/Controller.mo]: Don't we always use `Sta` for any eanble/disable status (unless it is in the name of the model)?

File: Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/FanSpeed/EnabledWSE/Subsequences/IntegratedOperation.mo [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+  Buildings.Controls.OBC.CDL.Logical.Switch fanSpe "Logical switch"
+    annotation (Placement(transformation(extent={{120,-90},{140,-70}})));
+  Buildings.Controls.OBC.CDL.Logical.Latch lat
+    "Logical latch, maintain ON signal until condition changes"
+    annotation (Placement(transformation(extent={{40,-120},{60,-100}})));
+  Buildings.Controls.OBC.CDL.Logical.Timer intOpeTim
+    "Count the time after plant switching from WSE-only mode to integrated operation mode"
+    annotation (Placement(transformation(extent={{80,-120},{100,-100}})));
+  Buildings.Controls.OBC.CDL.Continuous.GreaterEqualThreshold greEquThr(
+    final threshold=600)   "Check if it is 10 minutes after mode switch"
```
Comment (milicag) [inline Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/FanSpeed/EnabledWSE/Subsequences/IntegratedOperation.mo]: Please no values in comments

File: Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/FanSpeed/EnabledWSE/Subsequences/IntegratedOperation.mo [hunk 6]
```diff
@@ -0,0 +1,299 @@
...
+    annotation (Placement(transformation(extent={{80,90},{100,110}})));
+  Buildings.Controls.OBC.CDL.Continuous.MultiSum totMinCycLoa(final nin=nChi)
+    "Sum of minimum cyclining load for the operating chillers"
+    annotation (Placement(transformation(extent={{-20,90},{0,110}})));
+  Buildings.Controls.OBC.CDL.Continuous.MultiSum totLoa(final nin=nChi) "Total load of operating chillers"
+    annotation (Placement(transformation(extent={{-20,50},{0,70}})));
+  Buildings.Controls.OBC.CDL.Continuous.MultiSum minCycLoa(final nin=nChi)
+    "Sum of minimum cyclining load for all chillers"
+    annotation (Placement(transformation(extent={{-20,130},{0,150}})));
+  Buildings.Controls.OBC.CDL.Continuous.Division div "Output first input divided by second input"
```
Comment (milicag) [inline Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/FanSpeed/EnabledWSE/Subsequences/IntegratedOperation.mo]: I would rather implement this as it is described in 4.a). That means without the divisions. Also, in contrary of what the corresponding test shows, the fan speed should increase as the load increases above the minimum, in an attempt to increase the efficiency of the chiller by lowering the load.

File: Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/FanSpeed/EnabledWSE/Subsequences/Validation/WSEOperation.mo [hunk 7]
```diff
@@ -0,0 +1,94 @@
+within Buildings.Controls.OBC.ASHRAE.PrimarySystem.ChillerPlant.Tower.FanSpeed.EnabledWSE.Subsequences.Validation;
+model WSEOperation
+  "Validation sequence of controlling tower fan speed when waterside economizer is running alone"
```
Comment (milicag) [inline Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/FanSpeed/EnabledWSE/Subsequences/Validation/WSEOperation.mo]: Please check the consistency of the output of the test after revising the model tested as suggested.

File: Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/WaterLevel.mo [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+    annotation (Placement(transformation(extent={{100,-20},{140,20}}),
+      iconTransformation(extent={{100,-20},{140,20}})));
+
+protected
+  Buildings.Controls.OBC.CDL.Continuous.AddParameter addPar(
+    final p=watLevMin,
+    final k=-1)
+    "Minimum level minus current level"
+    annotation (Placement(transformation(extent={{-60,-10},{-40,10}})));
+  Buildings.Controls.OBC.CDL.Continuous.Hysteresis hys(
```
Comment (milicag) [inline Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/WaterLevel.mo]: See comment in line 22.

File: Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/WaterLevel.mo [hunk 9]
```diff
@@ -0,0 +1,84 @@
+within Buildings.Controls.OBC.ASHRAE.PrimarySystem.ChillerPlant.Tower;
+block WaterLevel
```
Comment (milicag) [inline Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/WaterLevel.mo]: Needs section number

File: Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/Staging/Subsequences/EnableCells.mo [hunk 10]
```diff
@@ -0,0 +1,1238 @@
+﻿within Buildings.Controls.OBC.ASHRAE.PrimarySystem.ChillerPlant.Tower.Staging.Subsequences;
+block EnableCells "Sequence for identifying enabing and disabling cells"
```
Comment (milicag) [inline Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/Staging/Subsequences/EnableCells.mo]: This model needs revision to reflect sections 2. and 3 more simply and generally. Please come over to discuss if you'd like.

**Comment by milicag** [comment]
@JayHuLBL This is ready for revisions.

File: Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/FanSpeed/EnabledWSE/Controller.mo [hunk 11]
```diff
@@ -0,0 +1,215 @@
+﻿within Buildings.Controls.OBC.ASHRAE.PrimarySystem.ChillerPlant.Tower.FanSpeed.EnabledWSE;
```
[the remark this answers is not in the document]
Comment (JayHuLBL) [reply Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/FanSpeed/EnabledWSE/Controller.mo]: I think it should be `EnableWSE` as the sequences in these package will be applied only when the WSE is enabled, either being enabled alone without chiller running, or both WSE and chiller are running.
Although it does have an output when the WSE is disabled, it is just zero.

File: Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/FanSpeed/EnabledWSE/Subsequences/IntegratedOperation.mo [hunk 12]
```diff
@@ -0,0 +1,299 @@
+﻿within Buildings.Controls.OBC.ASHRAE.PrimarySystem.ChillerPlant.Tower.FanSpeed.EnabledWSE.Subsequences;
+block IntegratedOperation
+  "Tower fan speed control when the waterside economizer is enabled and the chillers are running"
+
+  parameter Integer nChi = 2 "Total number of chillers";
+  parameter Modelica.SIunits.HeatFlowRate minUnLTon[nChi]={1e4, 1e4}
+    "Minimum cyclining load below which chiller will begin cycling";
+  parameter Real minSpe = 0.1
```
[the remark this answers is not in the document]
Comment (JayHuLBL) [reply Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/FanSpeed/EnabledWSE/Subsequences/IntegratedOperation.mo]: `fanSpeMin`

File: Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/FanSpeed/EnabledWSE/Subsequences/IntegratedOperation.mo [hunk 13]
```diff
@@ -0,0 +1,299 @@
...
+  parameter Modelica.SIunits.Time Td=0.1 "Time constant of derivative block"
+    annotation (Dialog(group="Load controller",
+                       enable=conTyp==Buildings.Controls.OBC.CDL.Types.SimpleController.PD or
+                              conTyp==Buildings.Controls.OBC.CDL.Types.SimpleController.PID));
+
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanInput uChi[nChi]
+    "Chiller enabling status: true=ON"
+    annotation (Placement(transformation(extent={{-200,80},{-160,120}}),
+      iconTransformation(extent={{-140,60},{-100,100}})));
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput chiLoa[nChi](
```
[the remark this answers is not in the document]
Comment (JayHuLBL) [reply Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/FanSpeed/EnabledWSE/Subsequences/IntegratedOperation.mo]: These has to be output from another block that does the calculation, or, if reading from the chiller if it has the meter in place.

File: Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/FanSpeed/EnabledWSE/Subsequences/IntegratedOperation.mo [hunk 14]
```diff
@@ -0,0 +1,299 @@
...
+    final unit=fill("W", nChi),
+    final quantity=fill("Power", nChi)) "Current load of each chiller"
```
[the remark this answers is not in the document]
Comment (JayHuLBL) [reply Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/FanSpeed/EnabledWSE/Subsequences/IntegratedOperation.mo]: `HeatFlowRate` seems to be more reasonable.

File: Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/FanSpeed/EnabledWSE/Subsequences/WSEOperation.mo [hunk 15]
```diff
@@ -0,0 +1,287 @@
...
+  Buildings.Controls.OBC.CDL.Continuous.GreaterEqualThreshold greEquThr(
+    final threshold=300)
+    "Check if fan is at minimum speed and the chilled water supply temperature drops below setpoint for 5 minutes"
+    annotation (Placement(transformation(extent={{80,90},{100,110}})));
+  Buildings.Controls.OBC.CDL.Continuous.Sources.Constant minTowSpe(
+    final k=minSpe) "Minimum tower speed"
+    annotation (Placement(transformation(extent={{-120,10},{-100,30}})));
+  Buildings.Controls.OBC.CDL.Continuous.Feedback feedback1 "Input difference"
+    annotation (Placement(transformation(extent={{-90,90},{-70,110}})));
+  Buildings.Controls.OBC.CDL.Continuous.Hysteresis hys2(
```
[the remark this answers is not in the document]
Comment (JayHuLBL) [reply Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/FanSpeed/EnabledWSE/Subsequences/WSEOperation.mo]: For the hold block, do you mean `TrueFalseHold`? It will be not sufficient here. The output is able to hold at lease 3 minutes, but cannot correctly switch back to true.
 
The condition (`CHWST rises above setpoint by 1 F`) for switching the output (fan status) back from `false` (cycle off) to `true` (turn on) is not the condition used for cycle off fan (`CHWST drop below setpoint and fans have been at minimum speed for 5 minutes`).

@milicag 
In fact, regarding the text `If CHWST drops below setpoint and fans have been at minimum fan speed for 5 minutes`, do you think it is 
`If (CHWST drops below setpoint and fans have been at minimum fan speed) for 5 minutes`, or
`If (CHWST drops below setpoint) and (fans have been at minimum fan speed for 5 minutes)`?

File: Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/FanSpeed/ReturnWaterTemperature/Subsequences/Enable.mo [hunk 16]
```diff
@@ -0,0 +1,390 @@
+﻿within Buildings.Controls.OBC.ASHRAE.PrimarySystem.ChillerPlant.Tower.FanSpeed.ReturnWaterTemperature.Subsequences;
+block Enable "Sequence for enabling and disabling tower fan"
```
Comment (milicag) [inline Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/FanSpeed/ReturnWaterTemperature/Subsequences/Enable.mo:2]: This is currently implemented for CWRT only? Just noticing that at some point it should have a flag to distinguish between when tower speed is controlled based on CWRT or CWST and be pulled up to a higher level in the package structure.

**Comment by milicag** [comment]
@JayHuLBL I've left several comments, mostly potential edits related to variable names and comments or clarification questions.

**Comment by milicag** [comment]
@JayHuLBL Can you please also add the variable name text to the top level controller icons for any of your packages top level controllers that still don't have it? Maybe we can merge that with this PR? I'll add it to mine on staging setpoints PR. This is useful for assembling and debugging the chiller plant controller. I saw you added it to the minimum flow bypass and it is really helpful.

File: Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/Controller.mo [hunk 17]
```diff
@@ -0,0 +1,507 @@
...
+          textString="%name")}),
+  Diagram(coordinateSystem(preserveAspectRatio=false, extent={{-100,-260},{100,260}})),
+Documentation(info="<html>
+<p>
+Block that controls cooling tower cells enabling status <code>yTowSta</code>, 
+the supply isolation valve positions <code>yIsoVal</code> of each cell and the 
+cell fan operating speed <code>yFanSpe</code>.
+This is implemented according to ASHRAE RP-1711 Advanced Sequences of Operation for 
+HVAC Systems Phase II – 
+Central Plants and Hydronic Systems (Draft 6 on July 25, 2019), section 5.2.12.
```
[the remark this answers is not in the document]
Comment (JayHuLBL) [reply Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/Controller.mo:475]: yes

File: Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/FanSpeed/EnabledWSE/package.mo [hunk 18]
```diff
@@ -0,0 +1,46 @@
+within Buildings.Controls.OBC.ASHRAE.PrimarySystem.ChillerPlant.Tower.FanSpeed;
+package EnabledWSE "Package of sequences for control tower fan speed when waterside economizer is enabled"
+
+annotation (
+preferredView="info", Documentation(info="<html>
+<p>
+This package contains sequences for cooling tower fan speed control when the waterside
+economizer is enabled.
+The implementation is based on section 5.2.12.2, item 4 in ASHRAE RP-1711, Draft 6 (July 25, 2019).
```
[the remark this answers is not in the document]
Comment (JayHuLBL) [reply Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/FanSpeed/EnabledWSE/package.mo:9]: yes

File: Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/Staging/Controller.mo [hunk 19]
```diff
@@ -0,0 +1,229 @@
...
+    annotation (Placement(transformation(extent={{100,40},{140,80}}),
+      iconTransformation(extent={{100,20},{140,60}})));
+  Buildings.Controls.OBC.CDL.Interfaces.RealOutput yIsoVal[nTowCel](
+    final unit=fill("1", nTowCel),
+    final min=fill(0, nTowCel),
+    final max=fill(1, nTowCel)) "Cooling tower cells isolation valve position"
+    annotation (Placement(transformation(extent={{100,-20},{140,20}}),
+      iconTransformation(extent={{100,-20},{140,20}})));
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanOutput yTowSta[nTowCel]
+    "Cooling tower cell enabling status"
```
[the remark this answers is not in the document]
Comment (JayHuLBL) [reply Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/Staging/Controller.mo:71]: In fact, I think your approach is better and I should change it align with yours, as "proven on" is more like guideline language.

File: Buildings/Resources/Scripts/Dymola/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/FanSpeed/ReturnWaterTemperature/Subsequences/Validation/Enable.mos [hunk 20]
```diff
@@ -0,0 +1,10 @@
...
+createPlot(id=1, position={10, 10, 1000, 1200}, y={"enaTow.uTow[1]"}, range={0.0, 3600, 0, 0.5}, grid=true, subPlot=3, colors={{28,108,200}});
+createPlot(id=1, position={10, 10, 1000, 1200}, y={"enaTow.yTow"}, range={0.0, 3600, 0, 0.5}, grid=true, subPlot=4, colors={{28,108,200}});
+createPlot(id=2, position={1020, 10, 1000, 1200}, y={"disTow.uMaxTowSpeSet[1]", "disTow.uMaxTowSpeSet[2]", "disTow.fanSpeMin"}, range={0.0, 3600, 0, 0.5}, grid=true, subPlot=1, colors={{28,108,200},{0,140,72},{238,46,47}});
+createPlot(id=2, position={1020, 10, 1000, 1200}, y={"disTow.yTow"}, range={0.0, 3600, 0, 0.5}, grid=true, subPlot=2, colors={{28,108,200}});
+createPlot(id=2, position={1020, 10, 1000, 1200}, y={"disTow1.uFanSpe","disTow1.fanSpeMin"}, range={0.0, 3600, 0, 0.5}, grid=true, subPlot=3, colors={{28,108,200},{238,46,47}});
+createPlot(id=2, position={1020, 10, 1000, 1200}, y={"disTow1.TTow", "disTow1.TTowSet"}, range={0.0, 3600, 0, 0.5}, grid=true, subPlot=4, colors={{28,108,200},{0,140,72}});
+createPlot(id=2, position={1020, 10, 1000, 1200}, y={"disTow1.yTow"}, range={0.0, 3600, 0, 0.5}, grid=true, subPlot=5, colors={{28,108,200}});
```
[the remark this answers is not in the document]
Comment (JayHuLBL) [reply Buildings/Resources/Scripts/Dymola/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/FanSpeed/ReturnWaterTemperature/Subsequences/Validation/Enable.mos:10]: I didn't see the empty line in mos script. However, I did you sometimes in the CI test, it may fail because last line is empty., but don't recall which file (order file? mos file? or model itself?)

File: Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/package.mo [hunk 21]
```diff
@@ -0,0 +1,45 @@
+within Buildings.Controls.OBC.ASHRAE.PrimarySystem.ChillerPlant;
+package Tower "Package of sequences for cooling tower control"
```
[the remark this answers is not in the document]
Comment (JayHuLBL) [reply Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/package.mo:2]: We may do it latter. I also noticed some other clean-up needed to be done, but we can do it in the staging branch.

File: Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/Staging/Subsequences/CellsNumber.mo [hunk 22]
```diff
@@ -0,0 +1,331 @@
...
+
+  Buildings.Controls.OBC.CDL.Interfaces.IntegerInput uChiSta
+    "Current chiller stage"
+    annotation (Placement(transformation(extent={{-300,80},{-260,120}}),
+      iconTransformation(extent={{-140,70},{-100,110}})));
+  Buildings.Controls.OBC.CDL.Interfaces.IntegerInput uChiStaSet
+    "Current chiller stage setpoint integer"
+    annotation (Placement(transformation(extent={{-300,40},{-260,80}}),
+      iconTransformation(extent={{-140,40},{-100,80}})));
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanInput uTowStaCha
```
[the remark this answers is not in the document]
Comment (JayHuLBL) [reply Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/Staging/Subsequences/CellsNumber.mo:28]: Yes

File: Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/Staging/Subsequences/StageProcesses.mo [hunk 23]
```diff
@@ -0,0 +1,393 @@
+﻿within Buildings.Controls.OBC.ASHRAE.PrimarySystem.ChillerPlant.Tower.Staging.Subsequences;
+block StageProcesses "Sequence for process of staging cells"
+
+  parameter Integer nTowCel = 4
+    "Total number of cooling tower cells";
+  parameter Real chaTowCelIsoTim(final quantity="Time", final unit="s") = 90
+    "Nominal time needed for open isolation valve";
+
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanInput uChaCel[nTowCel]
```
[the remark this answers is not in the document]
Comment (JayHuLBL) [reply Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/Staging/Subsequences/StageProcesses.mo:9]: This is a vector to flag if a cell should change its status, either being enabled or being disabled. I will change it to `Vector of boolean flags to show if a cell should change its status: true = the cell should change status (be enabled or disabled)`.

File: Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/FanSpeed/ReturnWaterTemperature/Subsequences/Coupled.mo [hunk 24]
```diff
@@ -0,0 +1,232 @@
...
+    annotation (Placement(transformation(extent={{-160,0},{-120,40}}),
+      iconTransformation(extent={{-140,0},{-100,40}})));
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput uMaxTowSpeSet[nChi](
+    final min=fill(0, nChi),
+    final max=fill(1, nChi),
+    final unit=fill("1", nChi))
+    "Maximum cooling tower speed setpoint from each chiller head pressure control loop"
+    annotation (Placement(transformation(extent={{-160,-40},{-120,0}}),
+      iconTransformation(extent={{-140,-40},{-100,0}})));
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanInput uChi[nChi]
```
[the remark this answers is not in the document]
Comment (JayHuLBL) [reply Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/FanSpeed/ReturnWaterTemperature/Subsequences/Coupled.mo:53]: proven on status

File: Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/FanSpeed/ReturnWaterTemperature/Subsequences/Enable.mo (hunk 16 above)
Comment (JayHuLBL) [reply Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/FanSpeed/ReturnWaterTemperature/Subsequences/Enable.mo:2]: Yes, this is just for CWRT. In fact, this sequence can be used for the control based on CWST. So I would prefer to leave (`TTow` and `TTowSet`) it like that for now.

Comment (milicag) [reply Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Tower/FanSpeed/ReturnWaterTemperature/Subsequences/Enable.mo:2]: Ok, it might be good to have that documented somewhere, either in the comment or in the info section just for future reference.

**Comment by JayHuLBL** [comment]
@milicag I have addressed your 1st round comments and added the connectors' name to the icon layer.

**Comment by milicag** [comment]
Thanks, I plan to take an another look at the functionality, and I'll merge it into 1900 (the chiller plant controller branch) to check for IO compatibility.

**Comment by milicag** [comment]
@JayHuLBL I plan to review this tomorrow morning.
